File: base/src/main/java/com/jointem/base/view/CustomToast.java
package com.jointem.base.view;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.jointem.base.R;

/**
 * @author JW.Lee
 * @ClassName: PayEasyToast
 * @Description: 自定义Toast
 * @date 2014年12月8日 上午10:55:56
 */
@SuppressLint("RtlHardcoded")
public class CustomToast {

    Context ctx;
    String str;

    int y;
    int x;

    public CustomToast(Context context, String string) {
        ctx = context;
        str = string;
        x = Gravity.LEFT;
        y = Gravity.TOP + 100;
    }

    public CustomToast(Context context, String string, int y1) {
        ctx = context;
        str = string;
        x = Gravity.LEFT;
        y = y1;
    }

    public CustomToast(Context context, String string, int x1, int y1) {
        ctx = context;
        str = string;
        x = x1;
        y = y1;
    }

    public CustomToast(Context context, int id) {
        ctx = context;
        this.str = context.getResources().getString(id);
        x = Gravity.LEFT + 50;
        y = Gravity.TOP + 100;
    }

    public CustomToast(Context context, int id, int y1) {
        ctx = context;
        this.str = context.getResources().getString(id);
        x = Gravity.LEFT + 50;
        y = y1;
    }

    public CustomToast(Context context, int id, int x1, int y1) {
        ctx = context;
        this.str = context.getResources().getString(id);
        x = x1;
        y = y1;
    }

    @SuppressLint({"InflateParams", "NewApi"})
    public void show() {
        Toast toast = new Toast(ctx);
        LayoutInflater inflater = (LayoutInflater) ctx
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View contentView = inflater.inflate(R.layout.toast, null);
        contentView.setBackground(ctx.getResources().getDrawable(R.drawable.search_box_toast));

        TextView text1 = (TextView) contentView.findViewById(R.id.tv_toast);
        text1.setText(str);
        toast.setView(contentView);
//		toast.setGravity(Gravity.BOTTOM, x, y);
        toast.setDuration(Toast.LENGTH_SHORT);
        toast.show();
    }

    @SuppressLint("InflateParams")
    public void show(int time) {
        Toast toast = new Toast(ctx);
        LayoutInflater inflater = (LayoutInflater) ctx
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View contentView = inflater.inflate(R.layout.toast, null);
        TextView text1 = (TextView) contentView.findViewById(R.id.tv_toast);
        text1.setText(str);
        toast.setView(contentView);
        toast.setGravity(Gravity.BOTTOM, x, y);
        toast.setDuration(time);
        toast.show();
    }

    public String getStr() {
        return str;
    }

    public void setStr(String str) {
        this.str = str;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }
}

File: app/src/main/java/com/joketng/base/view/CustomToast.kt
package com.joketng.base.view

import android.content.Context
import android.support.v4.content.ContextCompat
import android.support.v4.view.ViewCompat
import android.view.Gravity
import android.view.LayoutInflater
import android.widget.TextView
import android.widget.Toast
import com.joketng.base.R
import org.jetbrains.anko.find

/**
 * Created by thc on 2017/9/25.
 */
class CustomToast {

    var toast: Toast? = null
    lateinit var ctx: Context
    var str: String? = null
    var x = 0
    var y = 0

    constructor(ctx: Context, str: String?) {
        this.ctx = ctx
        this.str = str
        x = Gravity.LEFT
        y = Gravity.TOP + 100
    }


    constructor(ctx: Context, id: Int) {
        this.ctx = ctx
        this.str = ctx.resources.getString(id)
        x = Gravity.LEFT + 50
        y = Gravity.TOP + 100
    }

    fun show(time: Int = Toast.LENGTH_LONG) {
        if (toast == null) {
            toast = Toast(ctx)
        }
        val contentView = LayoutInflater.from(ctx).inflate(R.layout.toast, null)
        val drawable = ContextCompat.getDrawable(ctx, R.drawable.bg_toast)
        drawable?.alpha = 190
        ViewCompat.setBackground(contentView, drawable)
        val textView = contentView.find<TextView>(R.id.tv_toast)
        textView.text = str
        toast?.view = contentView
        toast?.duration = time
        toast?.show()
    }
}